### ISSUE TITLE
[Bug]: Input after 17 digit converts to `0` after loosing focus in Currency widget

### ISSUE THREAD

**Issue by chandannkumar** [appsmithorg/appsmith#20075]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

The inputs after 17 digits are getting converted to `0` automatically in Currency Input widget

### Steps To Reproduce

1. Drag & Drop Currency Input widget
2. Type in input with more 17 digits
3. Click outside of the currency widget
4. Observe that number after 17 digits got converted to `0`

### Issue video log

https://bthrujcsw8.vmaker.com/record/ladRprnBm6LCU28A

### Version

Cloud / Production

**Comment by MuhammadAakash** [comment]
Hi @chandannkumar 
I would like to work on this issue.

**Comment by dilippitchika** [comment]
Hello @MuhammadAakash can you please explain how you are thinking about the solution to this problem?

**Comment by MuhammadAakash** [comment]
Hi @dilippitchika 
First of all, I will figure out why this is happening, maybe there will be any type of validation of the length of numbers or the Currency widget is configured to only allow a certain number of digits.
If that is the case then i will remove that validation so it accepts if there are more than 17 digits in the input field

**Comment by MuhammadAakash** [comment]
@dilippitchika 
If you think there is any other better solution.
I would like to hear from you and then will start working according to that.

**Comment by dilippitchika** [comment]
@MuhammadAakash what do you think should be limit if you now remove the 17 digit limit?

**Comment by MuhammadAakash** [comment]
@dilippitchika 
I think no limit because it totally depends on the user that how many digits he/she likes to enter.

**Comment by dilippitchika** [comment]
Then i believe the problem will be in how large numbers are processed in JS. I think the 17 digit limit is already too large. Can you first investigate why there's a 17 digit limit?

**Comment by MuhammadAakash** [comment]
@dilippitchika  
Yeah sure I am cloning the repository, then I will start figuring out this limit.
But in JS The largest representable number is approximately 1.8 x 10^308, and the smallest is approximately 5 x 10^-324. Numbers outside of this range are considered "infinity" or "NaN" (not a number).

**Comment by MuhammadAakash** [comment]
@dilippitchika 
I have just visited the code and I found that where is the error
When the focus of input changes this function executes **handleFocusChange** and in this function, we are calling this function  **getLocaleThousandSeparator** to add a thousand separators in the value, 
`function getLocaleThousandSeparator() {
      return Intl.NumberFormat(getLocale())
        .format(11111)
        .replace(/\p{Number}/gu, "");
    }`
    
JavaScript's Number.MAX_SAFE_INTEGER is 9007199254740991 and **123456789123456789123** is much larger than that, so when passing this number to the Intl.NumberFormat will truncate the number and format it accordingly.

This means that any digits beyond the 17th digit will be truncated and not considered in the final value

**Comment by dilippitchika** [comment]
Thanks for the triaging on this @MuhammadAakash really appreciate it, i don't think we need to fix today this as it's very unlikely to have a currency with 17 digits without decimals. I suggest we park this feature for now.

**Comment by MuhammadAakash** [comment]
@dilippitchika 
Sure.

**Comment by dilippitchika** [comment]
Setting to low as people won't be facing this in a real scenario
